corehq/apps/locations/models.py

```diff
@@ -242,10 +244,33 @@ class SQLLocation(MPTTModel):
     # This should really be the default location manager
     active_objects = OnlyUnarchivedLocationManager()
 
+    @classmethod
+    def _migration_get_fields(cls):
+        return ["domain", "name", "lineage", "site_code", "external_id",
+                "metadata", "is_archived", "latitude", "longitude"]
+
+    @classmethod
+    def _migration_get_couch_model_class(cls):
+        return Location
+
+    def _migration_do_sync(self):
+        couch_obj = self._migration_get_or_create_couch_object()
+        couch_obj._sql_location_type = self.location_type
+        self._migration_sync_to_couch(couch_obj)
+
     def save(self, *args, **kwargs):
         from corehq.apps.commtrack.models import sync_supply_point
         self.supply_point_id = sync_supply_point(self)
-        return super(SQLLocation, self).save(*args, **kwargs)
+
+        sync_to_couch = kwargs.pop('sync_to_couch', True)
+        kwargs['sync_to_couch'] = False  # call it here
+        super(SQLLocation, self).save(*args, **kwargs)
```

Comment:
what is setting location_id on this model before the call to .save() here?  that's the only reason why for all the messaging stuff couch_id had to be nullable, because on the first time you .save() the sql model it doesn't have a couch id to link to